addons/mrp/models/mrp_production.py

```diff
@@ -1989,6 +1985,15 @@ def _default_amounts(production):
     def _action_confirm_mo_backorders(self):
         self.workorder_ids._action_confirm()
 
+    def _reproduce_same_sn(self):
+        for order in self:
+            mls = order.move_raw_ids.move_line_ids.filtered(
+                lambda ml: ml.lot_id and
+                    ml.lot_id == order.lot_producing_id and
+                        ml.move_id.product_id == order.product_id
+            )
+            mls.picked = True
```

Comment:
It's a very expensive process to check something that could barely happen (having the same serial in both component and output). I think we should be able to avoid such computation for all other manufacturing orders